packages/gatsby/src/datastore/in-memory/run-fast-filters.ts

```diff
@@ -295,7 +326,10 @@ export function runFastFiltersAndSort(args: IRunFilterArg): IQueryResult {
       ? sortedResult.slice(skip, limit ? skip + (limit ?? 0) : undefined)
       : sortedResult
 
-  return { entries: new GatsbyIterable(entries), totalCount }
+  const nodeObjects = entries
+    .map(nodeIds => getNode(nodeIds.id))
+    .filter(isGatsbyNode)
+  return { entries: new GatsbyIterable(nodeObjects), totalCount }
```

Comment:
should totalCount resemble nodeObjects instead of entries? Another question I have is, why do we have to do isGatsbyNode now and not before?

Reply:
> should totalCount resemble nodeObjects instead of entries?

I'm not sure what do you mean

> Another question I have is, why do we have to do isGatsbyNode now and not before?

`convertAndApplyFastFilters` now in master (few lines above) returns `Array<IGatsbyNode>`. With change to a `.map` that use `getNode`, before applying filter the type would be `Array<IGatsbyNode | undefined>` due to `getNode` return type being that way (if you call it with id that doesn't exist). Filter is there just for typescript pretty much. Note that we do basically same thing in

https://github.com/gatsbyjs/gatsby/blob/ec20301df296aeff5ec29f23d21ddfadd40032c1/packages/gatsby/src/datastore/lmdb/lmdb-datastore.ts#L132-L140

just here we use type guard instead of `Boolean` trick and avoid `as` casting when possible

Reply:
if it's just for typescript can we do an ! instead :) That would be a little bit cheaper.

> should totalCount resemble nodeObjects instead of entries?
Currently totalcount counts entries without the filter, so if it has undefined values they are not the same but as you mentioned this is just for typescript.